Service/Microsoft.Health.DeIdentification.Local/InMemoryQueueClient.cs

```diff
@@ -1,37 +1,106 @@
 ﻿using Microsoft.Health.JobManagement;
+using System.Collections.ObjectModel;
 
 namespace Microsoft.Health.DeIdentification.Local
 {
     public class InMemoryQueueClient : IQueueClient
     {
+        private int largestId = 1;
+        private List<JobInfo> jobInfos = new List<JobInfo>();
+        public List<JobInfo> JobInfos
+        {
+            get { return jobInfos; } 
+        }
+
         public Task CancelJobByGroupIdAsync(byte queueType, long groupId, CancellationToken cancellationToken)
         {
             throw new NotImplementedException();
         }
 
         public Task CancelJobByIdAsync(byte queueType, long jobId, CancellationToken cancellationToken)
         {
-            throw new NotImplementedException();
+            foreach (JobInfo jobInfo in jobInfos.Where(t => t.Id == jobId))
+            {
+                if (jobInfo.Status == JobStatus.Created)
+                {
+                    jobInfo.Status = JobStatus.Cancelled;
+                }
+
+                if (jobInfo.Status == JobStatus.Running)
+                {
+                    jobInfo.CancelRequested = true;
+                }
+            }
+
+            return Task.CompletedTask;
         }
 
-        public Task CompleteJobAsync(JobInfo jobInfo, bool requestCancellationOnFailure, CancellationToken cancellationToken)
+        public async Task CompleteJobAsync(JobInfo jobInfo, bool requestCancellationOnFailure, CancellationToken cancellationToken)
         {
-            throw new NotImplementedException();
+            JobInfo jobInfoStore = jobInfos.FirstOrDefault(t => t.Id == jobInfo.Id);
+            jobInfoStore.Status = jobInfo.Status;
```

Comment:
we need set status as [here](https://github.com/microsoft/FHIR-Analytics-Pipelines/blob/003c3a2ffb251bb1fd1dd1564bd135e445d8d360/FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.JobManagement/AzureStorageJobQueueClient.cs#L651)

Reply:
I will modify it